inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/sink/multiple/IcebergMultipleStreamWriter.java

```diff
@@ -201,7 +238,41 @@ public void processElement(RecordWithSchema recordWithSchema) throws Exception {
 
         if (multipleWriters.get(tableId) != null) {
             for (RowData data : recordWithSchema.getData()) {
-                multipleWriters.get(tableId).processElement(data);
+                try {
+                    multipleWriters.get(tableId).processElement(data);
+                } catch (Exception e) {
+                    LOG.error(String.format("write error, raw data: %s", data), e);
+                    if (!dirtyOptions.ignoreDirty()) {
+                        throw e;
+                    }
+                    if (dirtySink != null) {
+                        DirtyData.Builder<Object> builder = DirtyData.builder();
+                        try {
+                            builder.setData(data)
+                                    .setLabels(dirtyOptions.getLabels())
+                                    .setLogTag(dirtyOptions.getLogTag())
+                                    .setIdentifier(dirtyOptions.getIdentifier())
```

Comment:
label, logtag and identifier should be generated dynamicly when mutil sink

Reply:
@Yizhou-Yang review here

Reply:
parse the data by using DirtySinkhelper.replace(), and then jsondynamicschema.parse()